grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/DefaultGrpcExceptionHandlerFunction.java

```diff
@@ -0,0 +1,273 @@
+import io.grpc.Status.Code;
+import io.netty.handler.codec.http2.Http2Error;
+import io.netty.handler.codec.http2.Http2Exception;
+
+public final class DefaultGrpcExceptionHandlerFunction implements GrpcExceptionHandlerFunction {
+    private static final GrpcExceptionHandlerFunction INSTANCE = new DefaultGrpcExceptionHandlerFunction();
+
+    /**
+     * Returns the default {@link GrpcExceptionHandlerFunction}. This handler is also used as the final
+     * fallback when the handler customized
+     * with either {@link GrpcClientBuilder#exceptionHandler(GrpcExceptionHandlerFunction)}
+     * or {@link GrpcServiceBuilder#exceptionHandler(GrpcExceptionHandlerFunction)} returns {@code null}.
+     * For example, the following handler basically delegates all error handling to the default handler:
+     * <pre>{@code
+     * // For GrpcClient
+     * GrpcClients
+     *   .builder("http://foo.com")
+     *   .exceptionHandler((ctx, cause, metadata) -> null)
+     *   ...
+     *
+     * // For GrpcServer
+     * GrpcService
+     *   .builder()
+     *   .exceptionHandler((ctx, cause, metadata) -> null)
+     *   ...
+     * }</pre>
+     */
+    public static GrpcExceptionHandlerFunction ofDefault() {
+        return DefaultGrpcExceptionHandlerFunction.INSTANCE;
+    }
+
+    @Override
+    public @Nullable Status apply(RequestContext ctx, Throwable cause, Metadata metadata) {
+        return fromThrowable(cause);
+    }
+
+    /**
+     * Converts the {@link Throwable} to a {@link Status}, taking into account exceptions specific to Armeria as
+     * well and the protocol package.
+     */
+    public static Status fromThrowable(Throwable t) {
+        t = peelAndUnwrap(requireNonNull(t, "t"));
+        return statusFromThrowable(t);
+    }
+
+    /**
+     * Converts the {@link Throwable} to a {@link Status}.
+     * If the specified {@code statusFunction} returns {@code null},
+     * the built-in exception mapping rule, which takes into account exceptions specific to Armeria as well
+     * and the protocol package, is used by default.
+     */
+    public static Status fromThrowable(@Nullable GrpcStatusFunction statusFunction, RequestContext ctx,
+                                       Throwable t, Metadata metadata) {
+        final GrpcExceptionHandlerFunction exceptionHandler =
+                statusFunction != null ? statusFunction::apply : null;
+        return fromThrowable(exceptionHandler, ctx, t, metadata);
+    }
+
+    /**
+     * Converts the {@link Throwable} to a {@link Status}.
+     * If the specified {@link GrpcExceptionHandlerFunction} returns {@code null},
+     * the built-in exception mapping rule, which takes into account exceptions specific to Armeria as well
+     * and the protocol package, is used by default.
+     */
+    public static Status fromThrowable(@Nullable GrpcExceptionHandlerFunction exceptionHandler,
+                                       RequestContext ctx, Throwable t, Metadata metadata) {
```

Comment:
Should we remove both methods and their access? The caller should use the composed `GrpcExceptionHandlerFunction`.

Reply:
Applied feedback. Thanks